central/virtualmachine/service/service_impl.go

```diff
@@ -65,8 +73,31 @@ func (s *serviceImpl) GetVirtualMachine(ctx context.Context, request *v2.GetVirt
 }
 
 func (s *serviceImpl) ListVirtualMachines(ctx context.Context, request *v2.ListVirtualMachinesRequest) (*v2.ListVirtualMachinesResponse, error) {
-	// TODO: Filtering/search capabilities
-	vms, err := s.datastore.GetAllVirtualMachines(ctx)
+	searchQuery := search.EmptyQuery()
+	requestQuery := request.GetQuery().GetQuery()
+	if requestQuery != "" {
+		parsedQuery, err := search.ParseQuery(requestQuery)
+		if err != nil {
+			return nil, errors.Wrap(err, "parsing input query")
+		}
+		searchQuery = parsedQuery
+	}
+	paginated.FillPaginationV2(searchQuery, request.GetQuery().GetPagination(), defaultPageSize)
+	if len(searchQuery.GetPagination().GetSortOptions()) == 0 {
+		if searchQuery.GetPagination() == nil {
+			searchQuery.Pagination = &v1.QueryPagination{}
+		}
+		searchQuery.Pagination.SortOptions = []*v1.QuerySortOption{
+			{
+				Field: search.VirtualMachineName.String(),
+			},
+			{
+				Field: search.Namespace.String(),
+			},
```

Comment:
If you intend to fix the default sort to allow multiples fields, then write a ticket and add a TODO here.